### ISSUE TITLE
Problem sending BC

### ISSUE THREAD

**Issue by ross1701** [sparrowwallet/sparrow#1546]
Newbie here, so go easy please.
Any redirection to FAQs etc for this issue welcome.
So, essentially, I've tried to send some BC. The first transaction went fine. The second threw up some sort of error and is sitting as unconfirmed. I'll copy some of the log file below.

In laymans terms, what went wrong and how to do I fix it, please?

2024-10-21 15:46:47,302 ERROR [JavaFX Application Thread] c.s.s.p.ServerPreferencesController [null:-1] Connection error
com.sparrowwallet.sparrow.net.ElectrumServerRpcException: Error getting server version
	at com.sparrowwallet.sparrow@2.0.0/com.sparrowwallet.sparrow.net.SimpleElectrumServerRpc.getServerVersion(Unknown Source)
	at com.sparrowwallet.sparrow@2.0.0/com.sparrowwallet.sparrow.net.ElectrumServer.getServerVersion(Unknown Source)
	at com.sparrowwallet.sparrow@2.0.0/com.sparrowwallet.sparrow.net.ElectrumServer$ConnectionService$1.call(Unknown Source)
	at com.sparrowwallet.sparrow@2.0.0/com.sparrowwallet.sparrow.net.ElectrumServer$ConnectionService$1.call(Unknown Source)
	at javafx.graphics@22/javafx.concurrent.Task$TaskCallable.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at javafx.graphics@22/javafx.concurrent.Service.lambda$executeTask$6(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at javafx.graphics@22/javafx.concurrent.Service.lambda$executeTask$7(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: com.github.arteam.simplejsonrpc.client.exception.JsonRpcException: ErrorMessage{code=1, message=server.version already sent, data=null}
	at simple.json.rpc.client@1.3/com.github.arteam.simplejsonrpc.client.builder.RequestBuilder.executeAndConvert(Unknown Source)
	at simple.json.rpc.client@1.3/com.github.arteam.simplejsonrpc.client.builder.RequestBuilder.execute(Unknown Source)
	at com.sparrowwallet.sparrow@2.0.0/com.sparrowwallet.sparrow.net.SimpleElectrumServerRpc.lambda$getServerVersion$1(Unknown Source)
	at com.sparrowwallet.sparrow@2.0.0/com.sparrowwallet.sparrow.net.RetryLogic.getResult(Unknown Source)
	... 12 common frames omitted
2024-10-22 01:34:56,027 WARN [Thread-995] c.s.s.n.TcpTransport [null:-1] No response from server, setting read timeout to 8 secs
2024-10-22 01:35:05,988 WARN [Thread-995] c.s.s.n.TcpTransport [null:-1] No response from server, setting read timeout to 16 secs
2024-10-22 02:17:42,606 ERROR [JavaFX Application Thread] c.s.s.p.ServerPreferencesController [null:-1] Connection error
com.sparrowwallet.sparrow.net.ServerException: java.net.SocketException: Permission denied: connect
	at com.sparrowwallet.sparrow@2.0.0/com.sparrowwallet.sparrow.net.TcpTransport.connect(Unknown Source)
	at com.sparrowwallet.sparrow@2.0.0/com.sparrowwallet.sparrow.net.ElectrumServer.connect(Unknown Source)
	at com.sparrowwallet.sparrow@2.0.0/com.sparrowwallet.sparrow.net.ElectrumServer$ConnectionService$1.call(Unknown Source)
	at com.sparrowwallet.sparrow@2.0.0/com.sparrowwallet.sparrow.net.ElectrumServer$ConnectionService$1.call(Unknown Source)
	at javafx.graphics@22/javafx.concurrent.Task$TaskCallable.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at javafx.graphics@22/javafx.concurrent.Service.lambda$executeTask$6(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at javafx.graphics@22/javafx.concurrent.Service.lambda$executeTask$7(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.net.SocketException: Permission denied: connect
	at java.base/sun.nio.ch.Net.connect0(Native Method)
	at java.base/sun.nio.ch.Net.connect(Unknown Source)
	at java.base/sun.nio.ch.Net.connect(Unknown Source)
	at java.base/sun.nio.ch.NioSocketImpl.connect(Unknown Source)
	at java.base/java.net.SocksSocketImpl.connect(Unknown Source)
	at java.base/java.net.Socket.connect(Unknown Source)
	at java.base/sun.security.ssl.SSLSocketImpl.connect(Unknown Source)
	at java.base/sun.security.ssl.BaseSSLSocketImpl.connect(Unknown Source)
	at com.sparrowwallet.sparrow@2.0.0/com.sparrowwallet.sparrow.net.TcpOverTlsTransport.createSocket(Unknown Source)
	... 12 common frames omitted
2024-10-22 02:17:54,391 ERROR [JavaFX Application Thread] c.s.s.p.ServerPreferencesController [null:-1] Connection error
com.sparrowwallet.sparrow.net.ServerException: java.net.SocketException: Permission denied: connect
	at com.sparrowwallet.sparrow@2.0.0/com.sparrowwallet.sparrow.net.TcpTransport.connect(Unknown Source)
	at com.sparrowwallet.sparrow@2.0.0/com.sparrowwallet.sparrow.net.ElectrumServer.connect(Unknown Source)
	at com.sparrowwallet.sparrow@2.0.0/com.sparrowwallet.sparrow.net.ElectrumServer$ConnectionService$1.call(Unknown Source)
	at com.sparrowwallet.sparrow@2.0.0/com.sparrowwallet.sparrow.net.ElectrumServer$ConnectionService$1.call(Unknown Source)
	at javafx.graphics@22/javafx.concurrent.Task$TaskCallable.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at javafx.graphics@22/javafx.concurrent.Service.lambda$executeTask$6(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at javafx.graphics@22/javafx.concurrent.Service.lambda$executeTask$7(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.net.SocketException: Permission denied: connect
	at java.base/sun.nio.ch.Net.connect0(Native Method)
	at java.base/sun.nio.ch.Net.connect(Unknown Source)
	at java.base/sun.nio.ch.Net.connect(Unknown Source)
	at java.base/sun.nio.ch.NioSocketImpl.connect(Unknown Source)
	at java.base/java.net.SocksSocketImpl.connect(Unknown Source)
	at java.base/java.net.Socket.connect(Unknown Source)
	at java.base/sun.security.ssl.SSLSocketImpl.connect(Unknown Source)
	at java.base/sun.security.ssl.BaseSSLSocketImpl.connect(Unknown Source)
	at com.sparrowwallet.sparrow@2.0.0/com.sparrowwallet.sparrow.net.TcpOverTlsTransport.createSocket(Unknown Source)
	... 12 common frames omitted

And it goes on. Some sort of connection failure I get but specifically what and where?

**Comment by craigraw** [comment]
> Some sort of connection failure I get but specifically what and where?

Hard to say unfortunately. If you are using a public server, try a different one in the Server Preferences. If you are using a private server, you may need to restart it. Check your internet connection too.
